include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -384,23 +384,28 @@ __pattern_min_element(_ExecutionPolicy&& __exec, _Iterator __first, _Iterator __
         typename ::std::make_unsigned<typename ::std::iterator_traits<_Iterator>::difference_type>::type;
     using _ReduceValueType = tuple<_IndexValueType, _IteratorValueType>;
 
-    auto __identity_init_fn = __acc_handler_minelement<_ReduceValueType>{};
-    auto __identity_reduce_fn = [__comp](_ReduceValueType __a, _ReduceValueType __b) {
+    // This operator doesn't track the lowest found index in case of equal min. or max. values. Thus, this operator is
+    // not commutative.
+    auto __identity_reduce_fn = [__comp](_ReduceValueType __a, _ReduceValueType __b) -> _ReduceValueType {
```

Comment:
Here and in other lambdas, explicitly declaring the return type seems redundant, as it's deducible and also does not add much (if anything) for readability.

Reply:
Good point, I will change that. I think the only needed for `count` where we cast the return value of the predicate to an integer.